OrbitCore/TransactionManager.h

```diff
@@ -0,0 +1,98 @@
+#ifndef ORBIT_CORE_TRANSACTION_MANAGER_H
+#define ORBIT_CORE_TRANSACTION_MANAGER_H
+
+#include <memory>
+#include <queue>
+
+#include "Message.h"
+#include "Serialization.h"
+#include "Transaction.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/synchronization/mutex.h"
+
+namespace orbit {
+
+// The transaction manager is responsible for coordinating requests from the
+// client (UI) to the service and responses from the service back to the client.
+// The goal is to centralize communications between client and service to ensure
+// that only *one* transaction is in flight at at any given time. The user can
+// enqueue transactions having a guarantee that they will be executed in order.
+// Note that enqueueing requests is thread-safe.
+//
+// Usage: Register one TransactionHandler per message type. A TransactionHandler
+//        consists of a request handler that will be executed on the service
+//        side and a response handler for the client side (see Transaction.h).
+//        The steps for issuing requests and receiveing a response are:
+//
+//        1. The client enqueues a request through "EnqueueRequest".
+//        2. The TransactionManager schedules and sends out the request.
+//        3. Then service receives the request in its request handler.
+//           The request can be deserialized by calling "ReceiveRequest".
+//        4. The service sends a response through "SendResponse".
+//        5. The client receives the response in its response handler.
+//           The response can be deserialized by calling "ReceiveResponse".
+
+class TransactionManager {
+ public:
+  static TransactionManager& Get();
+  void RegisterTransactionHandler(const TransactionHandler& handler);
+
+  template <typename T>
+  inline static void EnqueueRequest(MessageType type, const T& object) {
+    Get().EnqueueRequestInternal(type, SerializeObjectBinary(object));
+  }
```

Comment:
explicitly delete T* variants so that we accidentally do not send pointers.

Reply:
Done.